extensions/wikia/Captcha/Captcha.setup.php

```diff
@@ -0,0 +1,177 @@
+ */
+$wgCaptchaStorageClass = 'Captcha\Store\Session';
+
+/**
+ * Allow users who have confirmed their e-mail addresses to post
+ * URL links without being harassed by the captcha.
+ */
+$wgAllowConfirmedEmail = false;
+
+/**
+ * Regex to whitelist URLs to known-good sites...
+ * For instance:
+ * $wgCaptchaWhitelist = '#^https?://([a-z0-9-]+\\.)?(wikimedia|wikipedia)\.org/#i';
+ * Local admins can define a whitelist under [[MediaWiki:captcha-addurl-whitelist]]
+ */
+$wgCaptchaWhitelist = false;
+
+/**
+ * Additional regexes to check for. Use full regexes; can match things
+ * other than URLs such as junk edits.
+ *
+ * If the new version matches one and the old version doesn't,
+ * toss up the captcha screen.
+ *
+ * @fixme Add a message for local admins to add items as well.
+ */
+$wgCaptchaRegexes = [];
+
+$dir = dirname( __FILE__ );
+$wgExtensionMessagesFiles['Captcha'] = "$dir/Captcha.i18n.php";
+
+// Captcha Hooks
+$wgHooks['EditFilterMerged'][] = 'Captcha\Hooks::confirmEditMerged';
+$wgHooks['UserCreateForm'][] = 'Captcha\Hooks::injectUserCreate';
+$wgHooks['AbortNewAccount'][] = 'Captcha\Hooks::confirmUserCreate';
+$wgHooks['LoginAuthenticateAudit'][] = 'Captcha\Hooks::triggerUserLogin';
+$wgHooks['UserLoginForm'][] = 'Captcha\Hooks::injectUserLogin';
+// $wgHooks['AbortLogin'][] = 'Captcha\Hooks::confirmUserLogin';
+$wgHooks['EmailUserForm'][] = 'Captcha\Hooks::injectEmailUser';
+$wgHooks['EmailUser'][] = 'Captcha\Hooks::confirmEmailUser';
+
+// Register API hook
+$wgHooks['APIEditBeforeSave'][] = 'Captcha\Hooks::confirmEditAPI';
+$wgHooks['APIGetAllowedParams'][] = 'Captcha\Hooks::APIGetAllowedParams';
+$wgHooks['APIGetParamDescription'][] = 'Captcha\Hooks::APIGetParamDescription';
+
+// General hooks
+$wgHooks['MakeGlobalVariablesScript'][] = 'Captcha\Hooks::onMakeGlobalVariablesScript';
+
+/**
+ * Autoload classes
+ */
+
+// Load Factories
+$wgAutoloadClasses['Captcha\Factory\Module'] = "$dir/Factory/Module.class.php";
+$wgAutoloadClasses['Captcha\Factory\Store'] = "$dir/Factory/Store.class.php";
+
+// Load captcha modules
+$wgAutoloadClasses['Captcha\Module\BaseCaptcha'] = "$dir/Module/BaseCaptcha.class.php";
+$wgAutoloadClasses['Captcha\Module\ReCaptcha'] = "$dir/Module/ReCaptcha.class.php";
+$wgAutoloadClasses['Captcha\Module\FancyCaptcha'] = "$dir/Module/FancyCaptcha.class.php";
+
+// Load storage modules
+$wgAutoloadClasses['Captcha\Store\Base'] = "$dir/Store/Base.class.php";
+$wgAutoloadClasses['Captcha\Store\Session'] = "$dir/Store/Session.class.php";
+$wgAutoloadClasses['Captcha\Store\Cache'] = "$dir/Store/Cache.class.php";
+
+// Main classes
+$wgAutoloadClasses['Captcha\Handler'] = "$dir/Captcha.class.php";
+$wgAutoloadClasses['CaptchaController'] = "$dir/CaptchaController.class.php";
+$wgAutoloadClasses['Captcha\Hooks'] = "$dir/CaptchaHooks.class.php";
+
+$wgCaptchaPrecedence = [
+	'Captcha\Module\ReCaptcha',
+	'Captcha\Module\FancyCaptcha',
+];
+
+// Always load the recaptcha shiz ... for now
```

Comment:
Just for reference, from this line to the end of the file should be deleted once https://github.com/Wikia/app/pull/6116 is merged.